### DISCUSSION TITLE
[REST API] Secp256k1 signature support

### DISCUSSION THREAD

**Discussion by Mama1emon** [aptos-labs/aptos-developer-discussions#106] [Questions]
Hello,

I am encountering an issue with sending a transaction from my secp256k1 address using the REST API. [The official documentation](https://aptos.dev/concepts/accounts#authentication-key) states that the blockchain supports this curve, but I am unable to find the signature type in [the API specification](https://api.devnet.aptoslabs.com/v1/spec#/).

**Comment by gregnazario** [comment]
It's made in a different way than Ed25519 signatures.  Instead they're under SingleKeySignature https://api.devnet.aptoslabs.com/v1/spec#/schemas/SingleKeySignature

And associated SingleKeyAuthenticator

**Reply by Mama1emon** [reply]
@gregnazario As per the specification, the request body format should be as follows:
```
"signature": {
    "type": "single_sender",
    "signature": {
        "type": "single_key_signature",
        "public_key": {
            "type": "secp256k1_ecdsa",
            "public_key": ""
        },
        "signature": {
            "type": "secp256k1_ecdsa",
            "signature": ""
        }
    }
}
```
However, I am encountering parsing errors with the request:
```
{
    "message": "parse request payload error: Expected input type \"AccountSignature\", found {\"type\":\"single_sender\",\"signature\":{\"type\":\"single_key_signature\",\"public_key\":{\"type\":\"secp256k1_ecdsa\",\"public_key\":\"\"},\"signature\":{\"type\":\"secp256k1_ecdsa\",\"signature\":\"\"}}}. (occurred while parsing \"TransactionSignature\") (occurred while parsing \"SubmitTransactionRequest\")",
    "error_code": "web_framework_error",
    "vm_error_code": null
}
```
Would it be possible for you to provide me with an example of the request body in JSON format that adheres to the specified format?

**Reply by Mama1emon** [reply]
@gregnazario

**Reply by Mama1emon** [reply]
@gregnazario

**Reply by gregnazario** [reply]
Hey, sorry was on some time off.

Getting some others to look at this.

**Reply by gregnazario** [reply]
Hey, it looks like there was a bug in the JSON format for single sender.

Highly suggest you use the BCS format in the TypeScript SDK, or is there a different language you prefer?

**Reply by Mama1emon** [reply]
@gregnazario
I'm using the Kotlin language, so I'm working with JSON.
Can you tell me when the bug will be fixed?

**Reply by davidiw** [reply]
The JSON API for transactions?

**Reply by gregnazario** [reply]
You can take a look at this Kotlin SDK https://github.com/mcxross/kaptos

I believe it has secp256k1 support via BCS
